examples/demo-rollup/src/txs_rpc.rs

```diff
@@ -0,0 +1,63 @@
+use jsonrpsee::RpcModule;
+// use serde::de::DeserializeOwned;
+// use serde::Serialize;
+use sov_rollup_interface::services::batch_builder::BatchBuilder;
+use sov_rollup_interface::services::da::DaService;
+use std::sync::{Arc, RwLock};
+
+/// Single data structure that manages mempool and batch producing.
+pub struct TxsRpcHandler<B: BatchBuilder, T: DaService> {
+    batch_builder: RwLock<B>,
+    da_service: Arc<T>,
+}
+
+impl<B: BatchBuilder + Send + Sync, T: DaService + Send + Sync> TxsRpcHandler<B, T> {
+    pub fn new(batch_builder: B, da_service: Arc<T>) -> Self {
+        Self {
+            batch_builder: RwLock::new(batch_builder),
```

Comment:
Do we need an `RwLock` here? It looks like we only ever call "write" - maybe a `Mutex` is more appropriate?